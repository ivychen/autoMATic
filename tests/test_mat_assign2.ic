/* Create a matrix */
// Assign to matrix literal
int main() {
    bool is_square;
		int matrix M = [[1,2,3];[3,2,1]];
    int matrix N;
    N = M';
    int i;
    int j;
    int test;
		is_square = true;
    i = 1;
    j = 1;

    print(M[0][0]); // 1
    print(M[1][0]); // 3
    print(M[0][1]); // 2
    print(M[1][1]); // 2
    print(N[1][0]); // 2


		if (is_square == true) {
      test = 10;
      print(M[1][1]); // 2
      M[0][0] = 5;
      print(M[0][0]); // 5
		} else {
      test = 20;
    }

    print(rows(N)); // 3
    print(cols(N)); // 2
    print(rows(M)); // 2
    print(cols(M)); // 3

    print(test); // 10
    test = M[i][j]; // set test = 2
    print(test); // 2
    print("success");
    return 0;
}
