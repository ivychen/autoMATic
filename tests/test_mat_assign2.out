1
3
2
2
2
2
5
3
2
2
3
10
2
success
